tests/ga/test_update.py

```diff
@@ -2868,171 +2868,6 @@ def test_it_should_process_goal_state_only_on_new_goal_state(self):
-        """
-        Creates an UpdateHandler instance that will run n iterations.
-
-        Can be supplied keyword args for:
-            *   mock_wire_data_file: This arg is treated like mock_wire_protocol
-                would. Defaults to mockwiredata.DATA_FILE_STATUS_BLOB
-            *   http_<action>_handler where action is get, put, or post: This arg
-                is treated like mock_wire_protocol would.
-        
-        Returned UpdateHandler instance has its protocol mocked via mock_wire_protocol.
-        """
-        
-        # Build the side_effect list for the UpdateHandler.is_running PropertyMock.
-        # Return True for the first n iterations followed by a single False to stop
-        # and then another True because the current UpdateHandler implementation
-        # does a __set__ during shutdown.
-        is_running_return_values = [True] * iterations + [False, True]
-        is_running_patch = patch.object(UpdateHandler, "is_running", PropertyMock(side_effect=is_running_return_values))
-
-        mock_http_get = kwargs.get("http_get_handler")
-        mock_http_put = kwargs.get("http_put_handler")
-        mock_http_post = kwargs.get("http_post_handler")
-
-        mock_wire_data_file = kwargs.get("mock_wire_data_file", mockwiredata.DATA_FILE_STATUS_BLOB)
-
-        try:
-            with mock_wire_protocol(mock_wire_data_file, mock_http_get, mock_http_post, mock_http_put) as protocol:
-                update_handler = get_update_handler()
-                
-                update_handler.protocol_util.get_protocol = Mock(return_value=protocol)
-
-                is_running_patch.start()
-                yield update_handler
-        finally:
-            is_running_patch.stop()
-
-    @staticmethod
-    def _fail_goal_state_fetch(url, **_):
-        """
-        For each goal state requested, returns values in order before failing with an
-        HttpError. Is useful for getting the agent into a specific state before causing
-        a failure.
-
-        Relies on this function to have the property return_vals populated with a list
-        of values to be returned in order. Any `None` in the list will cause the mock wire
-        data to be queried and returned, and thus functions as a sort of default.
-        """
-        if not HttpRequestPredicates.is_goal_state_request(url):
-            # url does not represent a request for a goal state; return None so
-            # that the mock_wire_protocol will return whatever data is in the mock
-            # wire data object (as per the mock_wire_protocol's docstring).
-            return None
-        try:
-            return ReportStatusTestCase._fail_goal_state_fetch.return_vals.pop()
-        except IndexError:
-            raise HttpError()
-
-    @patch("azurelinuxagent.common.conf.get_enable_fast_track", return_value=True)
-    def test_update_handler_should_report_status_even_on_failed_goal_state_fetch(self, _):
```

Comment:
This test and the one below have been problematic for several of the code iterations I've been working on.

I just went ahead and rewrote them as test_update_handler_should_report_status_when_fetch_goal_state_fails